File: tickezy_backend/src/models/payment.js
const { DataTypes } = require('sequelize');
const sequelize = require('../../db');

const Payment = sequelize.define('Payment', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true,
  },
  amount: {
    type: DataTypes.DOUBLE,
    allowNull: false,
  },
  paymentStatus: {
    type: DataTypes.ENUM('PENDING', 'SUCCESS', 'FAILED', 'REFUNDED'),
    defaultValue: 'PENDING',
  },
  paymentMethod: {
    type: DataTypes.ENUM('STRIPE', 'APPLE_PAY', 'CREDIT_CARD', 'MOBILE_MONEY', 'PAYPAL', 'CASH'),
    defaultValue: 'CREDIT_CARD',
  },
  paymentDate: DataTypes.DATE,
  transactionId: DataTypes.STRING,
}, {
  timestamps: true,
});

module.exports = Payment;

File: TICKEZY/TICKEZY/Models/Payment.swift
//
//  Payment.swift
//  TICKEZY
//
//  Created by M.A on 10/14/25.
//

import Foundation

enum PaymentStatus: String, Codable {
    case PENDING, SUCCESS, FAILED, REFUNDED
}

enum PaymentMethod: String, Codable {
    case STRIPE, APPLE_PAY
}

struct Payment: Codable, Identifiable {
    let id: String
    let userId: String
    let ticketId: String?
    let eventId: String
    let amount: Double
    let paymentStatus: PaymentStatus
    let paymentMethod: PaymentMethod
    let paymentDate: Date
    let transactionId: String
    let createdAt: Date
}

